### PR TITLE
Proof of concept of iDynTree native semantic aware geometric classes 

### PR DESCRIPTION
**Pull request by traversaro**
Proof of concept of iDynTree native semantic aware geometric classes. Semantic for the class is based on http://people.mech.kuleuven.be/~tdelaet/geometric_relations_semantics/geometric_relations_semantics_theory.pdf (the only small difference is that I removed a compose operation because it was ill-defined. 

For now only the `Position` class is provided, with binding in lua and matlab. 
There are several things that were implemented in this way but should be discussed, but I will open a issue on this.

### CODE REVIEW

**Comment by traversaro** [comment]
cc @francesco-romano @jeljaik

File: core-experimental/include/iDynTree/Core/Position.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        /**
+         * Copy constructor: create a Position from a PositionRaw
+         */
+        Position(const PositionRaw & other);
+
+        /**
+         * Denstructor
+         */
+        ~Position();
```
Comment (francesco-romano) [inline core-experimental/include/iDynTree/Core/Position.h]: Missing `virtual` here

**Comment by traversaro** [comment]
A brief tutorial on the new classes has been added in https://github.com/robotology-playground/idyntree/commit/8216fd857508d63e3f774fbcf8c2c0264920c7f5 . Merging otherwise no one will test.

File: core-experimental/include/iDynTree/Core/Position.h (hunk 1 above)
Comment (traversaro) [inline core-experimental/include/iDynTree/Core/Position.h]: Done in https://github.com/robotology-playground/idyntree/commit/088aabfc12cf2e6e039ea93acccdf731124a1e05 , thanks.

**Pull request merged**